pkg/asset/installconfig/powervs/metadata.go

```diff
@@ -30,63 +40,175 @@ func (m *Metadata) AccountID(ctx context.Context) (string, error) {
+
+	if m.accountID == "" {
+		apiKeyDetails, err := m.client.GetAuthenticatorAPIKeyDetails(ctx)
 		if err != nil {
 			return "", err
 		}
 
 		m.accountID = *apiKeyDetails.AccountID
 	}
+
 	return m.accountID, nil
 }
 
-// CISInstanceCRN returns the Cloud Internet Services instance CRN that is
-// managing the DNS zone for the base domain.
-func (m *Metadata) CISInstanceCRN(ctx context.Context) (string, error) {
+// APIKey returns the IBM Cloud account API Key associated with the authentication
+// credentials.
+func (m *Metadata) APIKey(ctx context.Context) (string, error) {
 	m.mutex.Lock()
 	defer m.mutex.Unlock()
 
-	if m.cisInstanceCRN == "" {
-		client, err := m.Client()
+	if m.client == nil {
+		client, err := NewClient()
 		if err != nil {
 			return "", err
 		}
 
-		zones, err := client.GetDNSZones(ctx)
+		m.client = client
+	}
+
+	if m.apiKey == "" {
+		m.apiKey = m.client.GetAPIKey()
+	}
+
+	return m.apiKey, nil
+}
+
+// GetCISInstanceCRN gets the CRN name for the specified base domain.
+func GetCISInstanceCRN(APIKey string, BaseDomain string) (string, error) {
+	var CISInstanceCRN string = ""
+	var bxSession *bxsession.Session
+	var err error
+	var tokenProviderEndpoint string = "https://iam.cloud.ibm.com"
+	var tokenRefresher *authentication.IAMAuthRepository
+	var authenticator *core.IamAuthenticator
+	var controllerSvc *resourcecontrollerv2.ResourceControllerV2
+	var listInstanceOptions *resourcecontrollerv2.ListResourceInstancesOptions
+	var listResourceInstancesResponse *resourcecontrollerv2.ResourceInstancesList
+	var instance resourcecontrollerv2.ResourceInstance
+	var zonesService *zonesv1.ZonesV1
+	var listZonesOptions *zonesv1.ListZonesOptions
+	var listZonesResponse *zonesv1.ListZonesResp
+
+	bxSession, err = bxsession.New(&bluemix.Config{
+		BluemixAPIKey:         APIKey,
+		TokenProviderEndpoint: &tokenProviderEndpoint,
+		Debug:                 false,
+	})
+	if err != nil {
+		return CISInstanceCRN, fmt.Errorf("getCISInstanceCRN: bxsession.New: %v", err)
+	}
+	tokenRefresher, err = authentication.NewIAMAuthRepository(bxSession.Config, &rest.Client{
+		DefaultHeader: gohttp.Header{
+			"User-Agent": []string{http.UserAgent()},
+		},
+	})
+	if err != nil {
+		return CISInstanceCRN, fmt.Errorf("getCISInstanceCRN: authentication.NewIAMAuthRepository: %v", err)
+	}
+	err = tokenRefresher.AuthenticateAPIKey(bxSession.Config.BluemixAPIKey)
+	if err != nil {
+		return CISInstanceCRN, fmt.Errorf("getCISInstanceCRN: tokenRefresher.AuthenticateAPIKey: %v", err)
+	}
+	authenticator = &core.IamAuthenticator{
+		ApiKey: APIKey,
+	}
+	err = authenticator.Validate()
+	if err != nil {
+		return CISInstanceCRN, fmt.Errorf("getCISInstanceCRN: authenticator.Validate: %v", err)
+	}
+	// Instantiate the service with an API key based IAM authenticator
+	controllerSvc, err = resourcecontrollerv2.NewResourceControllerV2(&resourcecontrollerv2.ResourceControllerV2Options{
+		Authenticator: authenticator,
+		ServiceName:   "cloud-object-storage",
```

Comment:
should this be something like cloud-internet-services instead of cloud-object-storage? that seems odd here